src/Storage/TSStorageManager.m

```diff
@@ -378,20 +442,29 @@ - (void)deleteThreadsAndMessages {
     [TSAttachmentStream deleteAttachments];
 }
 
-- (void)wipeSignalStorage {
-    self.database = nil;
-    NSError *error;
-
+- (void)deletePasswordFromKeychain {
     [SAMKeychain deletePasswordForService:keychainService account:keychainDBPassAccount];
-    [[NSFileManager defaultManager] removeItemAtPath:[self dbPath] error:&error];
-
+    [[NSUserDefaults standardUserDefaults] removeObjectForKey:TSStorageManagerHasDatabasePassword];
+    [[NSUserDefaults standardUserDefaults] synchronize];
+}
 
+- (void)deleteDatabaseFile {
+    NSError *error;
+    [[NSFileManager defaultManager] removeItemAtPath:[self dbPath] error:&error];
     if (error) {
         DDLogError(@"Failed to delete database: %@", error.description);
     }
+}
 
+- (void)wipeSignalStorage {
+    self.database = nil;
+    
+    [self deletePasswordFromKeychain];
+    
+    [self deleteDatabaseFile];
+    
     [TSAttachmentStream deleteAttachments];
-
+    
     [[self init] setupDatabase];
```

Comment:
Huh!  Also, why was this calling `init`?!?  I'm going to take your suggestion remove this line and undo the "splitting" I mentioned above.